sunpy/net/dataretriever/sources/noaa.py

```diff
@@ -148,61 +149,80 @@ class SRSClient(GenericClient):
     Results from 1 Provider:
     <BLANKLINE>
     2 Results from the SRSClient:
-           Start Time               End Time        Instrument ... Source Provider
-    ----------------------- ----------------------- ---------- ... ------ --------
-    2016-01-01 00:00:00.000 2016-01-02 00:00:00.000       SOON ...   SWPC     NOAA
-    2016-01-01 00:00:00.000 2016-01-02 00:00:00.000       SOON ...   SWPC     NOAA
+         Start Time           End Time      Instrument Physobs Source Provider
+    ------------------- ------------------- ---------- ------- ------ --------
+    2016-01-01 00:00:00 2016-01-01 23:59:59       SOON     SRS   SWPC     NOAA
+    2016-01-02 00:00:00 2016-01-02 23:59:59       SOON     SRS   SWPC     NOAA
     <BLANKLINE>
     <BLANKLINE>
 
     """
+    BASE_URL = 'ftp://ftp.swpc.noaa.gov/pub/warehouse/'
+    MIN_YEAR = 1996
 
     def _get_url_for_timerange(self, timerange):
         """
-        Returns a list of urls corresponding to a
-        given time-range.
+        Returns a list of urls corresponding to a given time-range.
         """
         result = list()
-        base_url = 'ftp://ftp.swpc.noaa.gov/pub/warehouse/'
-        total_days = int(timerange.days.value) + 1
-        all_dates = timerange.split(total_days)
-        today_year = int(Time.now().strftime('%Y'))
-        for day in all_dates:
-            end_year = int(day.end.strftime('%Y'))
-            if end_year > today_year or end_year < 1996:
-                continue
-            elif end_year == today_year:
-                suffix = '{}/SRS/{}SRS.txt'.format(
-                    end_year, day.end.strftime('%Y%m%d'))
-            else:
-                suffix = '{}/{}_SRS.tar.gz'.format(
-                    end_year, day.end.strftime('%Y'))
-            url = base_url + suffix
-            result.append(url)
-        return result
+        # Validate time range and return early if out side min/max
+        cur_year = Time.now().datetime.year
+        req_start_year = timerange.start.datetime.year
+        req_end_year = timerange.end.datetime.year
+        start_year = req_start_year if self.MIN_YEAR < req_start_year <= cur_year else None
+        end_year = req_end_year if self.MIN_YEAR < req_end_year <= cur_year else None
+        if start_year is None and end_year is None:
```

Comment:
If the start time or end time is out of range, but not both, then the start_year or end_year variables would be `None` and would presumably fail later in this method? Do we need to clip them to the MIN_YEAR / MAX_YEAR?

Reply:
Yea that didn't catch that kind of edge case reworked and added new test for these cases.